### PR TITLE
BIO_free should call method->destroy before free'ing any member fields

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Just a small logic fix.

### CODE REVIEW

File: crypto/bio/bio_lib.c [hunk 1]
```diff
@@ -98,6 +98,7 @@ int BIO_set(BIO *bio, const BIO_METHOD *method)
 
     bio->lock = CRYPTO_THREAD_lock_new();
     if (bio->lock == NULL) {
+        BNerr(BIO_F_BIO_SET, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/bio/bio_lib.c]: I think you mean `BIOerr`

Comment (FdaSilvaYY) [inline crypto/bio/bio_lib.c]: Fixed, thanks.

**Comment by levitte** [comment]
+1 (needs another team member review)

**Comment by t-j-h** [comment]
It is a behaviour change but it makes sense to me. 
+1 @levitte

**Comment by levitte** [comment]
Cool, thanks.  Merged!

**Pull request closed without merging**